include/istio/mixerclient/check_response.h

```diff
@@ -22,21 +22,15 @@
 namespace istio {
 namespace mixerclient {
 
-// The CheckResponseInfo holds response information in detail.
-struct CheckResponseInfo {
-  // Whether this check response is from cache.
-  bool is_check_cache_hit{false};
+// The CheckResponseInfo exposes policy and quota check details to the check
+// callbacks.
+class CheckResponseInfo {
+ public:
+  virtual ~CheckResponseInfo(){};
 
-  // Whether this quota response is from cache.
-  bool is_quota_cache_hit{false};
+  virtual const ::google::protobuf::util::Status& status() const = 0;
```

Comment:
Why rename `response_status()` to `status()`? This resulted in a lot of seemingly unnecessary changes in this PR.

Reply:
we discussed and decided it would have been good to do this in the first place, but is now unnecessary